WooCommerce/woocommerce_rms_normal/class-gateway.php

```diff
@@ -1,628 +1,874 @@
-    $this->payment_title = $this->settings['payment_title'];
-    $this->description = $this->settings['description'];
-    $this->merchant_id = $this->settings['merchant_id'];
-    $this->verify_key = $this->settings['verify_key'];
-    $this->secret_key = $this->settings['secret_key'];
-    $this->account_type = $this->settings['account_type'];
-    
-    // Define hostname based on account_type
-    $this->url = ($this->get_option('account_type')=='1') ? "https://pay.merchant.razer.com/" : "https://sandbox.merchant.razer.com/" ;
-    $this->inquiry_url = ($this->get_option('account_type')=='1') ? "https://api.merchant.razer.com/" : "https://sandbox.merchant.razer.com/" ;
-    
-    // Actions.
-    add_action( 'valid_molpay_request_returnurl', array( &$this, 'check_molpay_response_returnurl' ) );
-    add_action( 'valid_molpay_request_callback', array( &$this, 'check_molpay_response_callback' ) );
-    add_action( 'valid_molpay_request_notification', array( &$this, 'check_molpay_response_notification' ) );
-    add_action( 'woocommerce_receipt_molpay', array( &$this, 'receipt_page' ) );
-    
-    if (isset($_GET['molpay_redirect']) && $_GET['molpay_redirect']) {
-        $order_id = "";
-        if (isset($_GET['key']) && $_GET['key']) {
-            $order_id = wc_get_order_id_by_order_key( $_GET['key'] );
-            do_action ( "woocommerce_receipt_molpay", $order_id );
-            exit();
+class WC_Molpay_Gateway extends WC_Payment_Gateway
+{
+
+    // Constructor method
+    public function __construct()
+    {
+        global $woocommerce, $post;
+
+        $this->id                 = 'wcmolpay';
+        $this->icon = plugins_url('images/Fiuu_Logo-03.png', __FILE__);
+        $this->has_fields = false;
+        $this->method_title       = __('Fiuu', 'wcmolpay');
+        $this->method_description = __('Proceed payment via Fiuu Normal Integration Plugin', 'wcmolpay');
+        // Load the form fields.
+        $this->init_form_fields();
+
+        // Load the settings.
+        $this->init_settings();
+
+        // Define user setting variables.
+        $this->title = $this->settings['title'];
+        $this->ordering_plugin = $this->get_option('ordering_plugin');
+        $this->payment_title = $this->settings['payment_title'];
+        $this->description = $this->settings['description'];
+        $this->merchant_id = $this->settings['merchant_id'];
+        $this->verify_key = $this->settings['verify_key'];
+        $this->secret_key = $this->settings['secret_key'];
+        $this->account_type = $this->settings['account_type'];
+        $this->recurring = $this->settings['recurring'];
+
+        // Define hostname based on account_type
+        $this->url = ($this->get_option('account_type') == '1') ? "https://pay.fiuu.com/" : "https://sandbox.merchant.razer.com/";
+        $this->inquiry_url = ($this->get_option('account_type') == '1') ? "https://api.merchant.razer.com/" : "https://sandbox.merchant.razer.com/";
```

Comment:
change to api.fiuu.com

Reply:
changed

proof: 013ce67a70abd17bd3027c6e2410e84c1b2c025a